backend/api/test_credit_trades.py

```diff
@@ -483,3 +483,216 @@ def test_delete(self):
+            status='Approved')
+
+        completed_status, created = CreditTradeStatus.objects.get_or_create(
+            status='Completed')
+
+        cancelled_status, created = CreditTradeStatus.objects.get_or_create(
+            status='Cancelled')
+
+        draft_status, created = CreditTradeStatus.objects.get_or_create(
+            status='Draft')
+
+        credit_trade_type, created = CreditTradeType.objects.get_or_create(
+            the_type='Sell')
+
+        from_organization = Organization.objects.create(
+            name="Test 1",
+            actions_type_id=1,
+            status_id=1)
+        to_organization = Organization.objects.create(
+            name="Test 2",
+            actions_type_id=1,
+            status_id=1)
+
+        # the function shouldn't see this as it's only a draft and the
+        # initiator is not government
+        draft_credit_trade = CreditTrade.objects.create(
+            status=draft_status,
+            initiator=from_organization,
+            respondent=to_organization,
+            type=credit_trade_type,
+            number_of_credits=1000,
+            fair_market_value_per_credit=1,
+            zero_reason=None,
+            trade_effective_date=datetime.datetime.
+            today().strftime('%Y-%m-%d'))
+
+        # the function should see this as it's a draft from the government
+        draft_credit_trade_from_gov = CreditTrade.objects.create(
+            status=draft_status,
+            initiator=self.gov_user.organization,
+            respondent=to_organization,
+            type=credit_trade_type,
+            number_of_credits=1000,
+            fair_market_value_per_credit=1,
+            zero_reason=None,
+            trade_effective_date=datetime.datetime.
+            today().strftime('%Y-%m-%d'))
+
+        # the function shouldn't see this as it's approved
+        approved_credit_trade = CreditTrade.objects.create(
+            status=approved_status,
+            initiator=from_organization,
+            respondent=to_organization,
+            type=credit_trade_type,
+            number_of_credits=1000,
+            fair_market_value_per_credit=1,
+            zero_reason=None,
+            trade_effective_date=datetime.datetime.
+            today().strftime('%Y-%m-%d'))
+
+        # the function should see this as it's completed
+        completed_credit_trade = CreditTrade.objects.create(
+            status=completed_status,
+            initiator=from_organization,
+            respondent=to_organization,
+            type=credit_trade_type,
+            number_of_credits=1000,
+            fair_market_value_per_credit=1,
+            zero_reason=None,
+            trade_effective_date=datetime.datetime.
+            today().strftime('%Y-%m-%d'))
+
+        credit_trades = CreditTradeService.get_organization_credit_trades(
+            self.gov_user.organization
+        )
+
+        self.assertNotIn(draft_credit_trade, credit_trades)
```

Comment:
I would have broken each one of these into a separate test. I would create the CreditTradeStatus and CreditTradeType needed as part of the setup, and then have a separate test for each of the assertions.

Reply:
The problem with breaking these into separate tests is that it increases the chance of something going wrong. 

The aim of the test is to check and make sure that certain statuses show up and certain statuses to not show up when calling a specific function.

Reply:
Sure, there's always a trade-off like that when splitting up closely related tests, if you feel the trade-off isn't worth it, then it probably isn't.